### PR TITLE
[consensus] Remove unnecessary usage of bandwidth in network messages…

### PR DESCRIPTION
**Pull request by fxfactorial**
This should have a noticeable impact on our aws bill for network costs for bandwidth usage

### CODE REVIEW

File: consensus/validator.go [hunk 1]
```diff
@@ -172,17 +96,17 @@ func (consensus *Consensus) onPrepared(msg *msg_pb.Message) {
 		Uint64("MsgViewID", recvMsg.ViewID).
 		Msg("[OnPrepared] Received prepared message")
 
-	if recvMsg.BlockNum < consensus.blockNum {
+	if recvMsg.BlockNum != consensus.blockNum {
```
Comment (rlan35) [inline consensus/validator.go]: We should still store the blocks with Num > curBlock, in the case of missed messages or messages out of order. But for old blocks for sure we can ignore.

File: consensus/validator.go [hunk 2]
```diff
@@ -26,80 +24,7 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
...
-	header := new(block.Header)
-	if err := rlp.DecodeBytes(encodedHeader, header); err != nil {
-		consensus.getLogger().Warn().
-			Err(err).
-			Uint64("MsgBlockNum", recvMsg.BlockNum).
-			Msg("[OnAnnounce] Unparseable block header data")
-		return
-	}
-
-	if recvMsg.BlockNum < consensus.blockNum ||
```
Comment (rlan35) [inline consensus/validator.go]: why is block num verification also discarded? Then what's left in the announce?

**Comment by LeoHChen** [comment]
aws bandwidth is impacted by our pubsub gossiping, the size of the message is not a huge factor IMHO.

File: consensus/validator.go (hunk 2 above)
Comment (fxfactorial) [reply consensus/validator.go]: In unpushed code, I just didn't push yet. Have some more on the way

**Comment by fxfactorial** [comment]
@LeoHChen We'll find out for sure, soon enough :+1:

**Comment by LeoHChen** [comment]
> @LeoHChen We'll find out for sure, soon enough 👍

this has to be tested out before we merge it. we can actually find out the amount of data being transmitted using a smaller test net and do control group testing.  I am not comfortable of merging the code and test it on mainnet.

File: consensus/construct.go [hunk 3]
```diff
@@ -57,7 +57,6 @@ func (consensus *Consensus) construct(
 		}
 	case msg_pb.MessageType_COMMIT:
 		if s := consensus.priKey.SignHash(payloadForSignOverride); s != nil {
-			consensusMsg.Block = consensus.block
```
Comment (LeoHChen) [inline consensus/construct.go]: so we don't need to include the block in commit message? why there is no explanation in the commit message of about saving?

File: consensus/construct.go [hunk 4]
```diff
@@ -69,7 +68,7 @@ func (consensus *Consensus) construct(
 		buffer.Write(consensus.commitBitmap.Bitmap)
 		consensusMsg.Payload = buffer.Bytes()
 	case msg_pb.MessageType_ANNOUNCE:
-		consensusMsg.Payload = consensus.blockHeader
+		consensusMsg.Payload = consensus.blockHash[:]
```
Comment (LeoHChen) [inline consensus/construct.go]: why change this one?  please elaborate in the commit message.

File: consensus/construct.go (hunk 3 above)
Comment (LeoHChen) [reply consensus/construct.go]: was the block being used before? or it is totally just discarded before.

**Pull request merged**